src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -453,33 +452,29 @@ static int NPyAllSegOfSecIter_init_safe(NPyAllSegOfSecIter* self, PyObject* args
 }
 
 PyObject* NPySecObj_new(PyTypeObject* type, PyObject* args, PyObject* kwds) {
-    NPySecObj* self;
-    self = (NPySecObj*) type->tp_alloc(type, 0);
-    // printf("NPySecObj_new %p\n", self);
-    if (self != NULL) {
-        if (NPySecObj_init(self, args, kwds) != 0) {
-            Py_DECREF(self);
-            return NULL;
+    auto self = nb::steal(type->tp_alloc(type, 0));
+    // printf("NPySecObj_new %p\n", self.ptr());
+    if (self) {
```

Comment:
any reason not to `self && NPySecObj....` as opposed to nested if?

Reply:
Thank you, good point. No, but it might just be preparation for calling other C functions (that need return code checking). I don't really want to mix beautification with reference counting.